rotkehlchen/db/schema.py

```diff
@@ -338,10 +338,36 @@
 );
 """
 
+DB_CREATE_UNISWAP_EVENTS = """
+CREATE TABLE IF NOT EXISTS uniswap_events (
+    tx_hash VARCHAR[42] NOT NULL,
+    log_index INTEGER NOT NULL,
+    address VARCHAR[42] NOT NULL,
+    timestamp INTEGER NOT NULL,
+    type CHAR(1) NOT NULL,
```

Comment:
Hehehe ... I see what you tried to do here. But unfortunately that won't work.

You need to read this: https://www.sqlite.org/datatype3.html

> The affinity of a column is determined by the declared type of the column, according to the following rules in the order shown:
>
>1. If the declared type contains the string "INT" then it is assigned INTEGER affinity.
>
>2. If the declared type of the column contains any of the strings "CHAR", "CLOB", or "TEXT" then that column has TEXT affinity. Notice that the type VARCHAR contains the string "CHAR" and is thus assigned TEXT affinity.
> 
>3. If the declared type for a column contains the string "BLOB" or if no type is specified then the column has affinity BLOB.
>
>4. If the declared type for a column contains any of the strings "REAL", "FLOA", or "DOUB" then the column has REAL affinity.
> 
>5. Otherwise, the affinity is NUMERIC.

In sqlite CHAR essentially has text affinity so it's the same if you write `TEXT`. The only thing you can do is create an enum DB type like we do here:

https://github.com/rotki/rotki/blob/c0e01b71a8da1de2debb50e91f992add599013fe/rotkehlchen/db/schema.py#L2

And then reference that.

Or ... just save the `'mint'` and `'burn'` texts directly.

I will leave it to you to decide. No need to overcomplicate it. Just make sure to reflect the change in the code and the DB.

Reply:
I see, thanks. I'll go with `TEXT` column type. Unless we decide to abstract AMMs/LPs events, I don't think it is worth it to create a new table for referencing, as it would mean a new table per protocol.